iothub-manager/app/com/microsoft/azure/iotsolutions/iothubmanager/services/models/DeploymentServiceModel.java

```diff
@@ -23,55 +20,89 @@
+                    + ConfigurationsHelper.DEPLOYMENT_GROUP_ID_LABEL);
         }
 
-        if (!config.getLabels().containsKey(DEPLOYMENT_NAME_LABEL)) {
-            throw new InvalidInputException("Configuration is missing necessary label " + DEPLOYMENT_NAME_LABEL);
+        if (!deployment.getLabels().containsKey(ConfigurationsHelper.DEPLOYMENT_NAME_LABEL)) {
+            throw new InvalidInputException("Configuration is missing necessary label "
+                    + ConfigurationsHelper.DEPLOYMENT_NAME_LABEL);
         }
 
-        this.id = config.getId();
-        this.name = config.getLabels().get(DEPLOYMENT_NAME_LABEL);
+        this.id = deployment.getId();
+        this.name = deployment.getLabels().get(ConfigurationsHelper.DEPLOYMENT_NAME_LABEL);
 
-        String deviceGroupId = config.getLabels().get(DEPLOYMENT_GROUP_ID_LABEL);
+        String deviceGroupId = deployment.getLabels().get(ConfigurationsHelper.DEPLOYMENT_GROUP_ID_LABEL);
         String deviceGroupName = StringUtils.EMPTY;
-        if (config.getLabels().containsKey(DEPLOYMENT_GROUP_NAME_LABEL)) {
-            deviceGroupName = config.getLabels().get(DEPLOYMENT_GROUP_NAME_LABEL);
+        if (deployment.getLabels().containsKey(ConfigurationsHelper.DEPLOYMENT_GROUP_NAME_LABEL)) {
+            deviceGroupName = deployment.getLabels().get(ConfigurationsHelper.DEPLOYMENT_GROUP_NAME_LABEL);
         }
         this.deviceGroup = new DeviceGroup(deviceGroupId, deviceGroupName, null);
 
         this.packageName = StringUtils.EMPTY;
-        if (config.getLabels().containsKey(DEPLOYMENT_PACKAGE_NAME_LABEL)) {
-            this.packageName = config.getLabels().get(DEPLOYMENT_PACKAGE_NAME_LABEL);
+        if (deployment.getLabels().containsKey(ConfigurationsHelper.DEPLOYMENT_PACKAGE_NAME_LABEL)) {
+            this.packageName = deployment.getLabels().get(ConfigurationsHelper.DEPLOYMENT_PACKAGE_NAME_LABEL);
         }
 
-        this.createdDateTimeUtc =  this.formatDateTimeToUTC(config.getCreatedTimeUtc());
-        this.priority = config.getPriority();
-        this.type = DeploymentType.edgeManifest;
-        this.deploymentMetrics = new DeploymentMetrics(config.getSystemMetrics(), config.getMetrics());
+        this.createdDateTimeUtc =  this.formatDateTimeToUTC(deployment.getCreatedTimeUtc());
+        this.priority = deployment.getPriority();
+
+        if (deployment.getLabels().containsKey(ConfigurationsHelper.PACKAGE_TYPE_LABEL) &&
+            !(StringUtils.isBlank(deployment.getLabels().get(ConfigurationsHelper.PACKAGE_TYPE_LABEL))))
+        {
+            if (deployment.getLabels().containsValue(PackageType.edgeManifest.toString()))
+            {
+                this.packageType = PackageType.edgeManifest;
+            }
+            else if (deployment.getLabels().containsValue(PackageType.deviceConfiguration.toString()))
+            {
+                this.packageType = PackageType.deviceConfiguration;
+            }
```

Comment:
If it is neither maybe we should throw an exception?

Reply:
added